packages/cw1155/README.md

```diff
@@ -0,0 +1,48 @@
+# CW1155 Spec: Multiple Tokens
+
+CW1155 is a specification for managing multiple tokens based on CosmWasm.
+The name and design is based on Ethereum's ERC1155 standard.
+
+The specification is split into multiple sections, a contract may only
+implement some of this functionality, but must implement the base.
+
+## Base
+
+### Messages
+
+`TransferFrom{from, to, token_id, value}` - This transfers some amount of tokens between two accounts. The operator should have approval from the source account.
+
+Both `from` and `to` are `Option`, if `from` is `None`, it means mint new tokens, if `to` is `None`, it means burn tokens. They must not both be `None` at the same time.
```

Comment:
I now see the description here (I asked for it below). It would be good to have such essential docs also in the rustdoc on the types.

Having "transfer my tokens", "transfer your tokens", "mint new tokens", "burn my tokens" and "burn your tokens" all in the same message type does reduce some boilerplate, but would make it very easy to make mistakes. Especially since JSON ignores extra fields. (And the client sneds JSON)

If they accidentally use `rcpt` or `too` instead of `to`, this will burn the tokens. :scream: 

I see the idea of joining "send mine" and "send yours" as one type, but I would keep Mint and Burn as separate message types, so it is clear what is needed - and very unlikley to accidentally type "burn_from" instead of "transfer_from"